x/migration/module/cmd/import_morse_accounts.go

```diff
@@ -59,33 +62,41 @@ For more documentation, refer to: https://dev.poktroll.com/operate/morse_migrati
 func runImportMorseAccounts(cmd *cobra.Command, args []string) error {
 	ctx := cmd.Context()
 
-	// Ensure the morse account state file exists.
-	morseAccountsStateJSONPath := args[0]
-	if _, err := os.Stat(morseAccountsStateJSONPath); err != nil {
+	// Ensure the MsgImportMorseClaimableAccounts message JSON file exists.
+	msgImportMorseClaimableAccountsJSONPath := args[0]
+	if _, err := os.Stat(msgImportMorseClaimableAccountsJSONPath); err != nil {
 		return err
 	}
 
-	// Read and deserialize it the Morse account state file.
-	morseAccountStateBz, err := os.ReadFile(morseAccountsStateJSONPath)
+	// Read and deserialize it the MsrImportMorseClaimableAccounts message JSON file.
+	msgImportMorseClaimableAccountsJSONBz, err := os.ReadFile(msgImportMorseClaimableAccountsJSONPath)
 	if err != nil {
 		return err
 	}
 
-	// Unmarshal the Morse account state
-	morseAccountState := new(migrationtypes.MorseAccountState)
-	if err = cmtjson.Unmarshal(morseAccountStateBz, morseAccountState); err != nil {
+	// Deserialize the MsgImportMorseAccountState message.
+	msgImportMorseClaimableAccounts := new(migrationtypes.MsgImportMorseClaimableAccounts)
+	err = cmtjson.Unmarshal(msgImportMorseClaimableAccountsJSONBz, msgImportMorseClaimableAccounts)
+	if err != nil {
 		return err
 	}
 
-	// Construct a MsgImportMorseAccountState message.
-	msgImportMorseAccountState, err := migrationtypes.NewMsgImportMorseClaimableAccounts(
-		authtypes.NewModuleAddress(govtypes.ModuleName).String(),
-		*morseAccountState,
-	)
+	// Ensure the morse account state hash matches the expected morse account state hash.
+	expectedMorseAccountStateHash := msgImportMorseClaimableAccounts.GetMorseAccountStateHash()
+	actualMorseAccountStateHash, err := msgImportMorseClaimableAccounts.GetMorseAccountState().GetHash()
 	if err != nil {
 		return err
 	}
 
+	if !bytes.Equal(expectedMorseAccountStateHash, actualMorseAccountStateHash) {
```

Comment:
Why did we not need/have this before?

Reply:
Previously omitting this was not intentional but also not "necessary". I caught it this time around and classified it as "prudent".

Reply:
Noted - adding a comment.